racetrack_commons/racetrack_commons/api/asgi/error_handler.py

```diff
@@ -54,20 +47,60 @@ async def conflict_error_handler(request: Request, error: AlreadyExists):
         )
 
     @api.exception_handler(ValidationError)
-    async def validation_error_handler(request: Request, error: AlreadyExists):
+    def validation_error_handler(request: Request, error: AlreadyExists):
         """Validation Error"""
         return JSONResponse(
             content={"error": str(error), "type": type(error).__name__},
             status_code=400,  # Bad Request
         )
 
+    @api.exception_handler(Exception)
+    def default_error_handler(request: Request, error: Exception):
+        """Internal Server error"""
+        metric_internal_server_errors.inc()
+        log_request_exception_with_tracing(request, error)
+        error_message, error_type = _upack_error_message(error)
+        return JSONResponse(
+            status_code=500,
+            content={'error': error_message, "type": error_type},
+        )
+
     @api.middleware('http')
     async def catch_all_exceptions_middleware(request: Request, call_next):
         try:
             return await call_next(request)
+        except ExceptionGroup as e:
+            metric_internal_server_errors.inc()
+            log_request_exception_with_tracing(request, e)
+            error_message, error_type = _upack_error_message(e)
+            return JSONResponse(
+                status_code=500,
+                content={'error': error_message, "type": error_type},
+            )
         except BaseException as error:
+            metric_internal_server_errors.inc()
             log_request_exception_with_tracing(request, error)
             return JSONResponse(
                 status_code=500,
                 content={'error': str(error), "type": type(error).__name__},
             )
+
+
+def _upack_error_message(e: BaseException) -> tuple[str, str]:
+    if isinstance(e, ExceptionGroup):
+        sub_messages: list[str] = []
+        sub_types: list[str] = []
+        for suberror in e.exceptions:
+            sub_message, sub_type = _upack_error_message(suberror)
+            sub_messages.append(sub_message)
+            sub_types.append(sub_type)
+
+        if len(sub_messages) == 1:
+            e_message = sub_messages[0]
+            e_type = sub_types[0]
+        else:
+            e_message = f'{e}: ' + ', '.join(sub_messages)
+            e_type = f'{type(e).__name__}: ' + ', '.join(sub_types)
+        return e_message, e_type
+
+    return str(e), type(e).__name__
```

Comment:
In golang for sure the recommended thing to do would be to `if not isinstance(e,ExceptionGroup) return str, type` etc, and then the rest of logic without indentation. This gives flatter functions. I think here it also makes sense to follow that pattern.